chore(docs): refactor `bootstrap.recovery` section (#2550)
cloudnative-pg/cloudnative-pg: docs/src/bootstrap.md

used to spin up replica clusters. They both rely on the definition of external
clusters.

> Given the several possibilities, methods, and combinations that the

Reviewer: ```suggestion
Given there are several possible methods, and combinations that the
```

> CloudNativePG operator provides in terms of backup and recovery, please refer

Reviewer: ```suggestion
CloudNativePG operator provides, please refer
```

> to the ["Recovery" section](recovery.md).

Reviewer: ```suggestion
to the ["Recovery" section](recovery.md) for guidance for each method.
```

Reviewer: Just a suggestion. Haven't replicated it yet to the other locations of this text but...
